jupyter_server/serverapp.py

```diff
@@ -2508,6 +2509,11 @@ async def cleanup_extensions(self):
         self.log.info(extension_msg % n_extensions)
         await run_sync_in_loop(self.extension_manager.stop_all_extensions())
 
+    def cleanup_file_id_manager(self):
+        if not getattr(self, "file_id_manager", None):
+            return
+        self.file_id_manager._cleanup()
```

Comment:
I think if we add a `__del__` method to the FileIDManager, like this from the SessionManager, we don't need to add another cleanup method the main serverapp. It will be automatically cleaned up.

Reply:
Fixed here: https://github.com/jupyter-server/jupyter_server_fileid/commit/772e1a399b1c7cd797db2b3ef7d751042f0f6fe2